packages/web3-core-helpers/src/formatters.js

```diff
@@ -239,7 +239,14 @@ var outputTransactionFormatter = function (tx) {
         tx.transactionIndex = utils.hexToNumber(tx.transactionIndex);
     tx.nonce = utils.hexToNumber(tx.nonce);
     tx.gas = utils.hexToNumber(tx.gas);
-    tx.gasPrice = outputBigNumberFormatter(tx.gasPrice);
+    if (tx.gasPrice !== null)
```

Comment:
```suggestion
    if (tx.gasPrice)
```

After discussing with @GregTheGreek, we think that this `!== null` check is unnecessary, and should match the other if statements

Reply:
Agreed, I think my initial reasoning was if it was `0` then that check would fail and it wouldn't be formatted but it would be the string `"0x0"` in that case so your suggestion makes sense, will update.